clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsUbamTests.scala

```diff
@@ -378,4 +378,148 @@ trait WgsUbamTests { self: BaseIntegrationSpec =>
+        ClioCommand.moveWgsUbamName,
+        "--flowcell-barcode",
+        barcode,
+        "--lane",
+        lane.toString,
+        "--library-name",
+        library,
+        "--location",
+        Location.GCP.entryName,
+        "--destination",
+        cloudPath2.toUri.toString
+      )
+    } yield {
+      Files.exists(cloudPath) should be(false)
+      Files.exists(cloudPath2) should be(true)
+      new String(Files.readAllBytes(cloudPath2)) should be(fileContents)
+    }
+
+    result.andThen[Unit] {
+      case _ => {
+        val _ = Seq(cloudPath, cloudPath2).map(Files.deleteIfExists)
+      }
+    }
+  }
+
+  it should "not move wgs-ubams without a destination" in {
+    recoverToExceptionIf[Exception] {
+      runClient(
+        ClioCommand.moveWgsUbamName,
+        "--flowcell-barcode",
+        randomId,
+        "--lane",
+        "123",
+        "--library-name",
+        randomId,
+        "--location",
+        Location.GCP.entryName
+      )
+    }.map {
+      _.getMessage should include("--destination")
+    }
+  }
+
+  it should "delete wgs-ubams in GCP" in {
+    val barcode = s"abcdefg$randomId"
+    val lane = 4
+    val library = s"lib$randomId"
+
+    val fileContents = s"$randomId --- I am fated to die --- $randomId"
+    val cloudPath = rootTestStorageDir.resolve(
+      s"wgs-ubam/$barcode/$lane/$library/$randomId.unmapped.bam"
+    )
+    val deleteNote =
+      s"""Deleted by the integration tests, contents were:
+         |$fileContents
+       """.stripMargin
+
+    val key = TransferWgsUbamV1Key(barcode, lane, library, Location.GCP)
+    val metadata =
+      TransferWgsUbamV1Metadata(ubamPath = Some(cloudPath.toUri.toString))
+
+    // Clio needs the metadata to be added before it can be deleted.
+    val _ = Files.write(cloudPath, fileContents.getBytes)
+    val result = for {
+      _ <- runUpsertWgsUbam(key, metadata)
+      _ <- runClient(
+        ClioCommand.deleteWgsUbamName,
+        "--flowcell-barcode",
+        barcode,
+        "--lane",
+        lane.toString,
+        "--library-name",
+        library,
+        "--location",
+        Location.GCP.entryName,
+        "--note",
+        deleteNote
+      )
+      _ = Files.exists(cloudPath) should be(false)
+      response <- runClient(
+        ClioCommand.queryWgsUbamName,
+        "--flowcell-barcode",
+        barcode,
+        "--lane",
+        lane.toString,
+        "--library-name",
+        library,
+        "--location",
+        Location.GCP.entryName,
+        "--include-deleted"
+      )
+      outputs <- Unmarshal(response).to[Seq[TransferWgsUbamV1QueryOutput]]
+    } yield {
+      outputs should have length 1
+      outputs.head.notes should be(Some(deleteNote))
+      outputs.head.documentStatus should be(Some(DocumentStatus.Deleted))
+    }
+
+    result.andThen[Unit] {
+      case _ => {
+        val _ = Files.deleteIfExists(cloudPath)
+      }
+    }
+  }
+
+  it should "not delete wgs-ubams without a note" in {
```

Comment:
Are we also going to test that it appends instead of replacing notes?

Reply:
Yes we should, I'll add it.